src/openfermion/measurements/_probability_distributions.py

```diff
@@ -0,0 +1,379 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""
+Classes to store probability distributions for estimation
+of random variables.
+"""
+import numpy
+from scipy import sparse
+from numpy import cos, sin, pi
+from openfermion.config import EQ_TOLERANCE
+
+
+class FourierProbabilityDist(object):
+    """
+    Stores an approximate probability distribution of our knowledge of
+    a quantum state in the eigenbasis of a unitary operator.
+
+    A generic quantum state may be written in the eigenbasis of a unitary
+    operator U,
+
+    $|\psi\rangle = sum_j a_j |phi_j\rangle$,
+
+    where psi is the quantum state, |phi_j\rangle are the eigenstates of U,
+    labeled by their (unknown) eigenvalues
+
+    $U|phi_j\rangle = phi_j|phi_j\rangle$,
+
+    and a_j are the (unknown) corresponding amplitudes.
+
+    This class stores a distribution P(phi_0, phi_1,... , A_0, A_1,...)
+    (where A_j = |a_j|^2) of our knowledge of these variables.
+
+    In particular, it approximates this distribution as:
+    P(phi_0, phi_1,... , A_0, A_1,...)=P(A_0, A_1,...)*\prod_jP(\phi_j).
+
+    It then represents each phase distribution:
+    P(\phi_j) = sum_n (c_{j,2n} cos(n*phi_j) + c_{j,2n-1} sin(n*phi_j))
+
+    Separately, it stores an estimate of the mean amplitude \bar{A_j},
+    and data that allows for this to be recalculated (rather than a direct
+    representation of the function).
```

Comment:
Explain what you mean by "data that allows for this to be recalculated".

Reply:
So this is a bit of a hack, and honestly it's probably just better if I change it (the comment below is how I should be doing this). What I have been doing is running a maximum-likelihood estimator on the prior and a bunch of experimental data to get a good estimate of the mean amplitudes.

The full details of this are in the paper, but they're pretty sloppy and difficult to understand. As mentioned in the next comment, what I should be doing is getting rid of this part, and just updating a Normal distribution, similar to how the prior is passed. I'll get on this, but it'll take some time.